doc/cprover-manual/jbmc-user-manual.md

```diff
@@ -0,0 +1,237 @@
+to prove or refute properties. These properties are either automatically
+generated, or provided by the user in the form of assertions.
+
+Using examples, the following section will guide you through the basic
+features of JBMC, and explore the most useful features.
+
+
+\section jbmc-tutorial JBMC tutorial
+
+\subsection jbmc-manual-auto-assert Automatic assertions
+
+Consider the following simplistic `ExampleArray.java` file which we have compiled
+into `ExampleArray.class` and which contains a utility method to query a given
+array at a given index:
+```
+ 1| package tutorial;
+ 2|
+ 3| public class ExampleArray {
+ 4|
+ 5|     public static int query(int[] array, int index) {
+ 6|         return array[index];
+ 7|     }
+ 8| }
+```
+
+Now let's run the following command to let JBMC tell us about potential errors
+in our `query` method.
+
+    $ jbmc ExampleArray.class --function tutorial.ExampleArray.query
+
+
+The output contains the following:
+```c
+[java::tutorial.ExampleArray.query:([II)I.null-pointer-exception.1] line 6 Null pointer check: FAILURE
+[java::tutorial.ExampleArray.query:([II)I.array-index-out-of-bounds-low.1] line 6 Array index should be >= 0: FAILURE
+[java::tutorial.ExampleArray.query:([II)I.array-index-out-of-bounds-high.1] line 6 Array index should be < length: FAILURE
+[java::tutorial.ExampleArray.query:([II)I.1] line 6 no uncaught exception: SUCCESS
+```
+
+Three reported failures spring up:
+1. there is no null pointer check on the array passed as argument
+2. there is no check that the index at which we access the array is non-negative
+3. there is no check that the index is within the length of the array.
+
+If we make our code more robust by adding guards to make the array access
+safe as follows:
+```
+ 1| package tutorial;
+ 2|
+ 3| public class ExampleArraySafe {
+ 4|     public static int query(int[] array, int index) {
+ 5|         if (array == null) {
+ 6|             return -1;
+ 7|         }
+ 8|         if (index < 0 || index >= array.length) {
+ 9|             return -1;
+10|         }
+11|         return array[index];
+12|     }
+13|
+14| }
+```
+then the JBMC automatic assertions become valid, meaning that there is no
+possible inputs (argument values) for which they can be violated:
+```c
+[java::tutorial.ExampleArraySafe.query:([II)I.1] line 6 no uncaught exception: SUCCESS
+[java::tutorial.ExampleArraySafe.query:([II)I.null-pointer-exception.1] line 9 Null pointer check: SUCCESS
+[java::tutorial.ExampleArraySafe.query:([II)I.null-pointer-exception.2] line 12 Null pointer check: SUCCESS
+[java::tutorial.ExampleArraySafe.query:([II)I.array-index-out-of-bounds-low.1] line 12 Array index should be >= 0: SUCCESS
+[java::tutorial.ExampleArraySafe.query:([II)I.array-index-out-of-bounds-high.1] line 12 Array index should be < length: SUCCESS
+```
+
+\subsection jbmc-manual-unwind Loop unwinding
+
+As CBMC, JBMC transforms the input program into a state representation,
```

Comment:
I'm not sure 'state representation' is something intuitively understood by the reader.

Reply:
removing the sentence